<a href="https://colab.research.google.com/github/suryakanthPS/DS_Mini-Projects/blob/main/As3_Q4_Hypo_test_Customerorder.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#import libraries
import pandas as pd
import numpy as np
from scipy import stats as stats

* Null Hypothesis Ho: Independence of categorical variables (customer order does not vary by centers) 
*   Alternate Hypothesis as Ha: Dependence of categorical variables (customer order does vary and somewhat/significantly related)

In [ ]:
data = pd.read_csv("/content/Costomer+OrderForm.csv")
data.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [ ]:
data.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [ ]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [ ]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [ ]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [ ]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [ ]:
#make contingency table
obs= np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [ ]:
chi2=stats.chi2_contingency(obs)
chi2

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))



*   Result - pvalue 0.2771 is greater than sig lvl 0.05. Thus will accept null hypo
*   Inference - defective doesn't varies by centers

